pkg/network/setup/netstat.go

```diff
@@ -354,6 +360,11 @@ func updateVMIIfaceStatusWithGuestAgentData(ifaceStatus *v1.VirtualMachineInstan
 	// However, in case GA does not include IP data, it will clear IP status data (guest is not reachable by any IP).
 	ifaceStatusIPv4, ifaceStatusIPv6 := splitIPByFamiliy(ifaceStatus.IPs)
 	guestAgentIfaceIPv4, guestAgentIfaceIPv6 := splitIPByFamiliy(guestAgentIface.IPs)
+
+	if ifaceSpec.Masquerade != nil {
+		guestAgentIfaceIPv6 = filterLinkLocalAddresses(guestAgentIfaceIPv6)
```

Comment:
Please consider adding comment explaining why LLA should be filtered out on masquearde binding.

Reply:
Done.